drakcore/drakcore/app.py

```diff
@@ -154,6 +155,30 @@ def logindex(task_uid, log_type):
         return send_file(f.name)
 
 
+@app.route("/pcap_dump/<task_uid>")
+def pcap_dump(task_uid):
+    """
+    Return archaive containing dump.pcap along with extracted tls sessions
+    keys in format acceptable by wireshark.
+    """
+    analysis = AnalysisProxy(minio, task_uid)
+    try:
+        with NamedTemporaryFile() as f_pcap, NamedTemporaryFile() as f_keys, NamedTemporaryFile() as f_archive:
+            with ZipFile(f_archive, 'w', ZIP_DEFLATED) as archive:
+                analysis.get_pcap_dump(f_pcap)
+                archive.write(f_pcap.name, 'dump.pcap')
+                try:
+                    analysis.get_wireshark_key_file(f_keys)
+                    archive.write(f_keys.name, 'dump.keys')
+                except NoSuchKey:
+                    # No dumped keys.
+                    pass
+            f_archive.seek(0)
+            return send_file(f_archive.name, mimetype='application/zip')
```

Comment:
This is very linux-specific, i.e. we have a temporary file that will be open by flask (`send_file`), deleted (context manager exit) and then the file will be actually read.

@chivay could you evaluate whether this is OK or not?

Reply:
It is indeed correct. send_file opens the file under the hood so the file won't be reachable by name outside the context manager however we'll have a valid fd. :)

Reply:
:+1: thx